main.py

```diff
@@ -565,3 +548,25 @@ def handle_stats_reply(self, msg, switch):
             event = KytosEvent('amlight/flow_stats.flows_updated')
             event.content['switch'] = switch.dpid
             self.controller.buffers.app.put(event)
+
+    @listen_to('kytos/of_core.flow_stats.received')
+    def on_stats_received(self, event):
+        """Capture flow stats messages for OpenFlow 1.3."""
+        self.handle_stats_received(event)
+
+    def handle_stats_received(self, event):
+        """Handle flow stats messages for OpenFlow 1.3."""
+        switch = event.content['switch']
+        if 'replies_flows' in event.content:
+            replies_flows = event.content['replies_flows']
+            self.handle_stats_reply_received(switch, replies_flows)
+
+    def handle_stats_reply_received(self, switch, replies_flows):
+        """Iterate on the replies and set the generic flows"""
+        switch.generic_new_flows.extend(replies_flows)
+        switch.generic_flows = switch.generic_new_flows
+        switch.generic_flows.sort(
+            key=lambda f: (f.priority, f.duration_sec),
+            reverse=True
+        )
+        self.switch_stats_lock.setdefault(switch.id, Lock())
```

Comment:
@gretelliz this lock as it is seems unnecessary here, could you double check this? We could keep a similar usage https://github.com/kytos-ng/flow_stats/blob/master/main.py#L548-L549 to protect from unwanted preemption when setting the new flows and built the new list, but if you follow the prior suggestion of creating a list comprehension, and then also use the `sorted` flows on that new comprehension you'd be able to assign `switch.generic_flows` in a one-line statement, in that case I think we wouldn't even need the lock anymore since originally the lock helped out to also make sure the entire list was built correctly, which isn't a problem anymore.